### PR TITLE
docs(theming): added slider theming docs

### LINKED ISSUE #1087: Theming: Slider
### Subject

Slider

### Description

Create a basic theming documentation for the `Slider` component that showcases how you can style it.

Theming file: [Slider theming file](https://github.com/chakra-ui/chakra-ui-docs/blob/main/content/docs/components/slider/theming.mdx)

### Template Reference

For single part components, feel free to use the `Button` component theming doc as a template reference: [Button theming](https://github.com/chakra-ui/chakra-ui-docs/blob/main/content/docs/components/button/theming.mdx).

For multipart components, feel free to use the `Input` component theming doc as a template reference: [Input theming](https://github.com/chakra-ui/chakra-ui-docs/blob/main/content/docs/components/input/theming.mdx).

### PR DESCRIPTION
**Pull request by GKosheev**
Closes #1087 

## 📝 Description

> Adding ```Slider``` theming docs

## 📝 Additional Information

> SVG image needs to be generated & inserted

### CODE REVIEW

File: configs/sandpack-contents/component-theming/slider.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                  <SliderFilledTrack />
+                </SliderTrack>
+                <SliderThumb />
+              </Slider>
+            </Box>
+          </Flex>
+            </>
+          );
+        }
+      `,
```
Comment (aacevski) [inline configs/sandpack-contents/component-theming/slider.js:49]: Can you fix the formatting here? It looks bad on the CodeSandbox:
image

File: configs/sandpack-contents/component-theming/slider.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  Index: `import { ChakraProvider } from "@chakra-ui/react";
+  import * as React from "react";
+  import { createRoot } from "react-dom/client";
+  import App from "./App";
+  import "./styles.css";
```
Comment (aacevski) [inline configs/sandpack-contents/component-theming/slider.js:54]: Do we need these styles.css?

File: configs/sandpack-contents/component-theming/slider.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    }),
+  });
+  
+  export const sliderTheme = defineMultiStyleConfig({
+    baseStyle,
+    variants: { square },
+    sizes,
+  });  
+    `,
+}
```
Comment (aacevski) [inline configs/sandpack-contents/component-theming/slider.js:111]: There's formatting issues here as well:
image

File: configs/sandpack-contents/component-theming/slider.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        SliderThumb,
+        Text,
+        Box,
+        Flex,
+      } from "@chakra-ui/react";
+        
+        export default function App() {
+          return (
+            <>
+            <Flex textAlign="center" gap={3} mt={3} direction="column">
```
Comment (aacevski) [inline configs/sandpack-contents/component-theming/slider.js:15]: The examples are great, I just have one concern, the slider which is orange doesn't really go well with the blue, can you please change that?

Other than that, can you make the sliders have some padding so they don't take the whole screen.

And lastly if we can add the light/dark mode switch and I would add different styles when the dark mode is toggled.

**Review by aacevski — CHANGES_REQUESTED** [review]
Thanks a lot for your work in the theming, I highly appreciate it!

**Comment by GKosheev** [comment]
I think I've added everything needed. This PR still misses the ```SVG``` image of the slider's anatomy, so I can insert it as soon as it's ready.

File: configs/sandpack-contents/component-theming/slider.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+const baseStyle = definePartsStyle((props) => ({
+  filledTrack: {
+    bg: mode("blue.400", "blue.300")(props),
+  },
+  thumb: {
+    bg: mode("red.400", "red.200")(props),
+    _hover: {
+      bg: mode("red.500", "red.300")(props),
+    },
+  },
```
Comment (aacevski) [inline configs/sandpack-contents/component-theming/slider.js:102]: Hey! This part still hasn't been resolved. The light red doesn't really go well with the blue, can you pick an alternative color?

**Review by aacevski — CHANGES_REQUESTED** [review]
Slider Anatomy

And here's the Slider Anatomy!

File: configs/sandpack-contents/component-theming/slider.js (hunk 5 above)
Comment (GKosheev) [reply configs/sandpack-contents/component-theming/slider.js:102]: @aacevski, would white & gray work? 

```javascript
thumb: {
  bg: mode("white", "gray.200")(props),
  _hover: {
    bg: mode("gray.100", "gray.300")(props),
  },
```

Comment (aacevski) [reply configs/sandpack-contents/component-theming/slider.js]: Yep, looks good.

**Review by aacevski — APPROVED** [review]
This can be merged, we just need to fix conflicts.

**Comment by GKosheev** [comment]
@aacevski, I've just resolved all the conflicts. I have another PR with ```Popover```, so I'll use your recommendations to improve the Showcase section 🙂

**Comment by aacevski** [comment]
> @aacevski, I've just resolved all the conflicts. I have another PR with `Popover`, so I'll use your recommendations to improve the Showcase section 🙂

I meant the branch cannot be merged due to conflicts:
image

**Comment by GKosheev** [comment]
@aacevski, it seems I don't have permission for that. Is there any other way to fix this? This file is automatically updated every time I commit changes.

image

**Comment by aacevski** [comment]
> @aacevski, it seems I don't have permission for that. Is there any other way to fix this? This file is automatically updated every time I commit changes.
> 
> image

You will need to sync your fork with the main repository like this:
image

The next step is to head to your terminal / editor and do:
- Checkout to the main branch by running: `git checkout main`
- Pull the latest changes:  `git pull`
- Checkout to your branch: `git checkout slider-theming-docs`
- Merge the main branch into yours `git merge main`
- Resolve the conflicts by just running `pnpm run search-meta:gen`
- And commit your changes.

**Pull request merged**